### PR TITLE
[wasm] asset loading for workers

### PR DESCRIPTION
**Pull request by pavelsavara**
- move asset related logic to separate file`assets.ts`
- split logging code into separate file `logging.ts`
- add `dotnet.wasm` and `dotnet-crypto-worker.js` to mono-config.json via `WasmAppBuilder`
- moved `readSymbolMapFile()` call to `startup.ts`
- fixed bug in `setup_proxy_console()` which didn't really copy the original `log` and `error` functions of the console and caused recursion on error.
- moved `init_crypto` after `mono_wasm_load_config` in `mono_wasm_pre_init_essential_async`
- added new resource type `js-module-crypto` and `js-module-threads`
- changed the detection of blazor startup sequence to not be disabled by presence of config, but only by config.assets having some assembly in it.

Most of this changes were moved from https://github.com/dotnet/runtime/pull/73073 where is log of discussions about it
Together with https://github.com/dotnet/aspnetcore/pull/43049

### CODE REVIEW

File: src/mono/wasm/runtime/startup.ts [hunk 1]
```diff
@@ -584,297 +452,70 @@ export function bindings_init(): void {
...
+        afterConfigLoaded.promise_control.resolve();
         return;
     }
     if (runtimeHelpers.diagnosticTracing) console.debug("MONO_WASM: mono_wasm_load_config");
     try {
         const resolveSrc = runtimeHelpers.locateFile(configFilePath);
         const configResponse = await runtimeHelpers.fetch_like(resolveSrc);
         const configData: MonoConfig = (await configResponse.json()) || {};
+        if (configData.environmentVariables && typeof (configData.environmentVariables) !== "object")
+            throw new Error("Expected config.environment_variables to be unset or a dictionary-style object");
```
Comment (maraf) [inline src/mono/wasm/runtime/startup.ts:478]:
```suggestion
            throw new Error("Expected config.environmentVariables to be unset or a dictionary-style object");
```

**Review by maraf — APPROVED** [review]
LGTM

**Comment by pavelsavara** [comment]
Unfortunately the deadlock is still with us. [Log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-73484-merge-7d07af3bd6eb411baa/WasmTestOnBrowser-subtlecrypto-System.Security.Cryptography.Tests/1/console.a90a81fa.log?helixlogtype=result)

**Comment by radical** [comment]
You can run the tests a good bunch(30?) of times on helix, which can be done in one shot as we had discussed offline. It is helpful in finding similar issues.

I wonder if we should always do that for tests that depend on workers, on CI.

**Comment by pavelsavara** [comment]
- OSX issue is unrelated
 - Installer issue is unrelated: [Log](https://dev.azure.com/dnceng/public/_build/results?buildId=1931495&view=logs&jobId=2de479a8-eae7-5324-c49d-bebaef329288&j=2de479a8-eae7-5324-c49d-bebaef329288&t=bb577e75-8247-55f1-4164-04f3b0f2d1eb)
 - debugger issue is https://github.com/dotnet/runtime/issues/72946 [Log](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-73484-merge-0ce80faad137472e89/chrome-DebuggerTests/1/console.92d9ae5f.log?helixlogtype=result)

**Comment by radical** [comment]
The wasm/debugger test failures are unrelated, and will be fixed by https://github.com/dotnet/runtime/pull/73524 .

**Comment by radical** [comment]
We should queue up the crypto deadlock fix right after this. If this happens to cause any issue with that, then it can be fixed in that PR.

**Pull request merged**